### ISSUE TITLE
Mobile Theme disappears on reopening project

### ISSUE THREAD

**Issue by wocmultimedia** [maqetta/maqetta#3391]
Hi, 

I created a test mobile project with 3 views then following the video on youtube I created a new Mobile Theme and applied to my project. I tested on both Chrome and Firefox last v. and it worked. Saved and turned off the PC. The following day I launched again maqetta server bat and then open the localhost in chrome, the project charged automatically but in the designer window all the mobile silhouette was a mess picture not track of my theme. I clicked on preview and it shows correctly the project. Back to design window and I couldn't switch theme it said no theme available. I tried to open the theme editor but message say no user theme found. I work on XP Pro SP3 system what do I have to do?
I already get some problem with the mobile theme and reinstalled maqetta from zero, this is the second time I install it. Of course I cleaned the browser cache several times before writing this issue.
Thanks for help.

**Comment by doronrosenberg** [comment]
This is with M7 correct?

**Comment by wocmultimedia** [comment]
Yes, Sir!
Besides, I installed first time and in Chrome it didn't worked correctly the my mobile theme giving false colours while in FF I could see the colours. I reinstalled and it worked on chrome but the day later on restarting the PC happened what I told you. I can add a screen shot here of the server.bat and preview of the project1 but I cannot add a screen shot of the Designer cause I cannot access again Maqetta, it stops at login interface. I also tried to signup again how a new user but nothing.

  ----- Original Message -----

**Comment by JonFerraiolo** [comment]
-> @billreed63 Can you please try to recreate?

**Comment by wocmultimedia** [comment]
Hi John,

Magically this morning I succeeded to login into the maqetta designer interface and you can see the screenshot of the mess picture related to the preview shot I enclose again.
You can see on the right doesn't appear the corresponding widget settings....I'm on Chrome latest version working on XP.

Thanks for helping
Angelo
  ----- Original Message -----

**Comment by wocmultimedia** [comment]
Maybe this shot can be useful too, if I try to switch the theme, always appears this prompt to save but there's nothing to save in the project and after clicking save it switch to the panel  but nothing happen if I select a different theme iphone, etc.

Regards,
Angelo
  ----- Original Message -----

**Comment by wocmultimedia** [comment]
I recreated it again and it happens again. I Created in the same project a new mobile application. Created 3 views and then make a new theme. I set new colour values and named it different than previous theme. I applied and saved the project. It worked fine till I decided to make some new value changes to the theme. I set the new values then applied and nothing happened. Ok, I reloaded the page refreshing the browser and again the silhouette in the designer view lost all references to widget and theme while preview is shown with widget.

**Comment by wocmultimedia** [comment]
I Recreated it and found when it happen!!! If I create a project in the "root" it happen while if the project is created with the "Eclipse support" option checked and it is created in the WebContent it works correctly. I can refresh or reload and it is still working and showing correctly either in Designer or Preview. I also restart the PC and the root project still appear a mess silhouette while other project in WebContent is correct.

**Comment by billreed63** [comment]
Could you add the html source of the problem file to the issue so I can attempt to reproduce?

**Comment by wocmultimedia** [comment]
Hi sorry for my maybe stupid reply, where do I find the html file created by maqetta? or can I copy the source code in an empty html file send it to you?

I start adding now the copied source code then if isn't correct please tell me where do I can find what you need to analize it.
Thanks
angelo
  ----- Original Message -----

**Comment by billreed63** [comment]
You can cut and past the source from  the source editor in maqetta. To add the source to the issue use the the four of the  " ` "  on my keyboard it is on the same key as ~ http://github.github.com/github-flavored-markdown/

**Comment by wocmultimedia** [comment]
I try to add as you told me

```
<!DOCTYPE html>
<html>
<head>
<meta charset="utf-8"/>
<title>Untitled</title>
<script type="text/javascript" src="lib/dojo/dojo/dojo.js" data-dojo-config="'parseOnLoad':true,'async':true,'packages':[{'name':'gridx','location':'../gridx'},{'name':'clipart','location':'../../clipart'},{'name':'maqettaSamples','location':'../../../samples'},{'name':'maqetta','location':'../../maqetta'},{'name':'shapes','location':'../../shapes'},{'name':'zazl','location':'../../zazl'},{'name':'widgets','location':'../../custom'}],'themeMap':[['Android',null,['themes/Theme2/Theme2.css']],['BlackBerry',null,['themes/Theme2/Theme2.css']],['iPad',null,['themes/Theme2/Theme2.css']],['iPhone',null,['themes/Theme2/Theme2.css']],['.*',null,['themes/Theme2/Theme2.css']]],'mblThemeFiles':[], mblLoadCompatPattern: /\/project1\/themes\/.*\.css$/"></script>
<script type="text/javascript">
require([
  "dijit/dijit",
  "dojo/parser",
  "maqetta/space",
  "maqetta/AppStates",
  "dojox/mobile",
  "dojox/mobile/deviceTheme",
  "dojox/mobile/compat",
  "dojo/hash",
  "shapes/Circle"
]);
</script>
<style>@import "themes/claro/claro.css";@import "lib/shapes/shapes.css";@import "app.css";
</style>
<script type="text/javascript" src="app.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=no"></meta>
</head>
<body data-maq-device="iphone" data-maq-flow-layout="true" data-maq-ws="collapse" class="claro" id="myapp" data-maq-appstates="{}">
 <div id="View_0" data-dojo-type="dojox.mobile.View" style="height: 100%; display: none;">
  <h1 label="Norway Guide" data-dojo-type="dojox.mobile.Heading"></h1>
  <ul data-dojo-type="dojox.mobile.RoundRectList">
    <li data-dojo-type="dojox.mobile.ListItem" moveTo="View_2" rightIcon="mblDomButtonArrow" transition="flip">
  Info</li>
    <li data-dojo-type="dojox.mobile.ListItem" transition="swirl" moveTo="View_3">
      Video</li>
  </ul>
</div>
 <div id="View_2" data-dojo-type="dojox.mobile.View" style="height: 100%; display: none;">
  <h1 label="Info Page" data-dojo-type="dojox.mobile.Heading" back="home" moveTo="View_0"></h1>
  <div style="font-weight: bold;">
    Just a few notes to test this view 1</div>
</div>
 <div id="View_3" data-dojo-type="dojox.mobile.View" style="height: 100%;" selected="selected">
  <h1 label="Video List" data-dojo-type="dojox.mobile.Heading" back="home" moveTo="View_0" transition="fade"></h1>
  <div style="font-weight: bold;">
    Here will appear a Video List about Norwegian places</div>
  <div>
Another piece of text</div>
  <textarea style="width: 240px; height: 81px;">
This has to be another colour yellow pale the background and Blue text</textarea>
<span data-dojo-type="shapes.Circle" rx="74.5" ry="53" style="position: absolute; z-index: 900; left: 76px; top: 223px; width: 170px; height: 95px; background-color: transparent;" cx="37.25" cy="26.5"></span></div>
</body>
</html>
```

**Comment by billreed63** [comment]
I don't see any problems in the HTML, when you see the problem in the designer what happens if close the tab that the file is in and then reopen the file?
Could you also send a screen capture. you can not add images to issues in github you will have to post the image to a photo sharing site like http://imgur.com/ and then add a link to the image to the issue

**Comment by wocmultimedia** [comment]
http://www.wocmultimedia.com/blogstore/maqetta/maqettaErrorInDesigner.jpg
http://www.wocmultimedia.com/blogstore/maqetta/maqettaprojectdesigner.jpg
http://www.wocmultimedia.com/blogstore/maqetta/maqettaprojectdesigner2.jpg
http://www.wocmultimedia.com/blogstore/maqetta/maqettaprojectpreview.jpg

I wish the url works you can see 2 different designer project silhouette file1.html and file2.html and the preview of file.html.....closing any tab making any attempt nothing change also cleaning cache, monitoring on FF besides Chrome and restarting PC or reloading the page it is always the same while a project in WebContent is ok.

**Comment by billreed63** [comment]
This looks like the timing issue that we have been having since we changed the code to use async loading
If when you see these errors you can preview the page and it is displayed correctly it is most likely the timing problem
in chrome try opening the Tools->Developer tools and refresh your browser, in most cases this may changing the timing enough so the issue is resolved. If that is the case then we can work more on the timeing

**Comment by wocmultimedia** [comment]
Yes, the preview was correct as you can se from the screen shot but now I cannot open any project anymore....I login and maqetta automatically access last project that was one of this not working but shows only the general display like when you create a new project. I try to open or access any project and the buttons open but nothing happen. No other screen than this appear, you can see something is missing on the left column above outline.
Please, watch the shot http://www.wocmultimedia.com/blogstore/maqetta/maqettaErrorNoProject.jpg 
in my humble opinion is Java 7 the problem cause in many software including Eclipse Indigo java 7 creates problems and I have to point it at previous version but, of course I'm not a true developer just a newbie

**Comment by wocmultimedia** [comment]
Hello, good and bad news.
I succeeded to start again maqetta going to C:\Maqetta_7\maqetta\$absusersdir.metadata.plugins\org.eclipse.orion.server.core.settings\Users.prefs then I changed the line about Active Editor....
Maqetta restarted and then I followed your suggestion to open Tools>Developer and refresh the browser. It closed the server, I restarted the server.bat then relogin in Maqetta and it shows correct the silhouette...So, You have to work more on this code direction as you told me....Thanks, I wait for some fix because in this way cannot be employed in some production.

**Comment by billreed63** [comment]
That could very well be, I do not know of any testing on Java 7. Most of the team is running java 6 (1.6)  I personally have only tested on Java 1.6.0_25-b06

**Comment by wocmultimedia** [comment]
What is strange is that in the new requirements is stated Java 7 on your main page, maybe because of new dojo 1.8 (?) I don't know...If is better to use Java 6 how can I inform Maqetta server to use Java 6 instead of Java 7? sorry, I'm a filmmaker not a developer so it's difficult for me understand many things.
Thanks

**Comment by billreed63** [comment]
Did you download Maqetta from this url http://maqetta.org/downloads/ and which version? Where did you see java 7 as a requirement?

**Comment by wocmultimedia** [comment]
Yes, but I'm not sure where I read it cause I scrolled so many pages in these days so I can be confused or I read on Dojotoolkit website. If Maqetta works better with java 6 I suggest you to add a system requirement page so we can set it properly. How can I try to use Java 6 with Maqetta? Is there any command or line to add in some places?
Thanks.
Angelo

**Comment by billreed63** [comment]
Dup of #3329

**Comment by wocmultimedia** [comment]
Thanks! I wish you succeed to solve it in M8 or in a next version of M7 to help also common people like me to create nice product with your very interesting software.

**Comment by wocmultimedia** [comment]
I think this other strange behaviour is connected to previous one but it's up to you understand it.
I created a new project populated with 3 views and a scrolling_view
I add a picture to the Menu_Item background and I load a .png image for the same button linking to a View using moveTo:... I downloaded the workspace complete. Then I tested the files and result is on this shot were is the picture shown

http://www.wocmultimedia.com/blogstore/maqetta/maq7PictureShown1.jpg

I turned off the PC and restart later but opening the project the picture is not shown anymore eithier in Designer view or Preview mode. Both pictures are still there on the images folder as before. See shot below. 

http://www.wocmultimedia.com/blogstore/maqetta/maq7PictureNoshown.jpg

Thanks for helping

**Comment by JonFerraiolo** [comment]
@wocmultimedia: do you have the ability to pull down nightly builds from http://maqetta.org/nightly/ ? (Don't use the "hosted' version, use the other one) It would be nice to know if the problem has gone away in the nightly builds. We hope to ship a minor update with this fix within the next few days.

**Comment by wocmultimedia** [comment]
I downladed the file. How do I have to use it? My M7 is set in C:\Maqetta7\maqetta 
do you mean I've tu build another project or where I find the project created if I install this new version in another folder?
Sorry, but I'm not a so clever developer
Thanks
Angelo

**Comment by wocmultimedia** [comment]
Hi Jon,
I uploaded a screen shot of my local disk where M7 is installed. I noted something strange between M6 and M7 configuration about " user" folder. On M6 in the user folder appears the username (my email) which I still use to access and the project1 is listed while in the same folder on M7 it's totally empty. Maybe is normal but I prefer to show it to you because I haven't found my projects in no folder. This is the screen shot 
http://www.wocmultimedia.com/blogstore/maqetta/maqettalocationondisk.jpg
Thanks

**Comment by wocmultimedia** [comment]
@billreed63
It was a bit difficult to find the lines you suggested in the #3391 cause no number correspondence in my M7 even though using search function in notepad++ I succeeded to add these lines in places you showed but nothing changed at all, my pictures are always not shown.
Sorry, you seems annoyed to reply to such problem but please, your fix #3391 isn't aspirin to be used for any symptoms. I didn't say "theme" isn't reloaded but inside the view pictures are not shown.

@JonFerraiolo where do I find the folder with my projects  in the local disk? Just to copy them and test in the other nightly build M7 you suggested me.
Thanks both for the support.

**Comment by billreed63** [comment]
Have you downloaded the nightly build and been able to reproduce the issue on the nightly build? 
If the nightly build fixes your issue then you will need to apply all the commits in this issue to M7. or better yet replace your M7 with the nightly.

**Comment by wocmultimedia** [comment]
I downloaded and installed the September 26, nightly build but I need to know where I can find the projects folder to open the problematic project with the new version.
In my local disk I'm not able to find any folder related to projects created in the M7 you can see by yourself that screen shot doesn't show any project in M7 user while there is project1 in M6 folder. 
If you can say me where projects are stored I can test soon, thanks

**Comment by billreed63** [comment]
The users files are now stored in the folder  `$absusersdir` If you look in that folder you will most likely see letters for example `a` look in those folders and you will see the files from your workspace. on the nightly build create a user first then copy the files from M7 to the nightly  for example from M7 folder `a` copy the contents to nightly build folder `a`.  

I really hate to ask you this but could you download and try the 9/28 nightly as it has the most recent updates?

**Comment by jhpedemonte** [comment]
@wocmultimedia, if you downloaded a Maqetta build and started it up using the included scripts, then you will have a folder called **users** in the Maqetta directory.  This is where all files and user information is stored.

To use the same data for a new build, simply copy that **users** folder to the new build and start from the scripts.

**Comment by wocmultimedia** [comment]
Ok I will download and try it why not, I'm here to go on with you not only for my needs

**Comment by wocmultimedia** [comment]
Here I'm!
I made several test how below described
Nightly build 26/9
Created a new similar project to that not correct shown and it worked, I turned off and restart PC it is still there working.
I downloaded and installed 28/9 nightly 
then I went in old M7 $abusersdir found that the folder was "j" (no mention about project name, I guessed by the custom folder I added)  I copied and pasted on the new 28/9 $abusersdir but nothing happen, the program never show the project at all even reloading or restarting.
I created 10 empty projects with one view each (there's I believe a bug, if you don't set at least one view the create new project doesn't work properly after 2 projects go back to project one and shows only 2 projects)
When I reached the new created folder "j" I went in older folder and copied again the files inside this new "j", in this case Maqetta saw soon the project on reloading. The results is that:
- The older project inside 28/9 nightly is still without pictures on both Design and Preview windows.
- The similar project created with 26/9 is shown correctly also here.

Not satisfied, I exported this newest project as complete workspace then I shut the browser and restart the PC.
Again, the old project doesn't show the pictures (they still are in the same folder with the same name of newest project)
while the new similar project is correctly shown.

It's up to you now to deduct how to proceed about. What can be useful is a way to know how to recognize the project name in the $abusersdir.

Thanks, I wish I've been useful.

**Comment by billreed63** [comment]
It sounds like your theme files are not being loaded, check to ensure that your have copied the theme  that your file is using over to your new workspace.. For example if the page that is not displaying the theme is using Theme2 use the open -> theme to open Theme2 to ensure that it is indeed in the workspace.  If it is indeed in the workspace try switching the theme of the of the problem page to android theme for all devices and seem if that fixes the issue.

**Comment by billreed63** [comment]
For mapping projects read this post https://groups.google.com/forum/?fromgroups=#!topic/maqetta-devs/B5HQov9o80w

**Comment by wocmultimedia** [comment]
Ok, I'm not disappeared I just spent 1 hour making tests cause I went out of office and closed the browser then, on my way back, I restarted chrome and M7 but the old project hanged. I tried to restart the PC and to re-paste the old folder but nothing at all. All other projects opened regularly at any new restart but not the old one.
Here's a link about the server message during the hanging phase I suppose there's something you can check but one part of the sentence says "Ignore extra content ok".
I opened the old project only in it's old M7 version always without pictures, I changed the mobile theme to all using the different themes android, and so on but nothing. I cannot make more tests cause the project never open in M7 nigthly neither 26 nor 28.
Here's the screen shot http://www.wocmultimedia.com/blogstore/maqetta/M7serverwindow.jpg I wish it's helpful.
Thanks

**Comment by wocmultimedia** [comment]
@billreed63 Your latest link is very helpful for my question about Projects and other references, Thanks a lot.

**Comment by billreed63** [comment]
If you can create a new project and open a new mobile page with the correct display on the 28 nightly build then you have a problem with the project you are trying to copy over.

**Comment by wocmultimedia** [comment]
Yes, billreed63, I tested in any sausage! In fact, during the hanging ( I mean M7continues loading for several minutes) I can click and open Files tab and select another existing project of the newly created that opens regularly.

**Comment by billreed63** [comment]
Since M701 is working correctly I would suggest that you copy tour files over to a project folder you have created in M701. I would only copy the HTML and Themes.  One at time to see which file is corrupted.

**Comment by wocmultimedia** [comment]
I will let you know about, later. Thanks

---------- Header Originale -----------

Da      : "Bill Reed" notifications@github.com
A          : "maqetta/maqetta" maqetta@noreply.github.com
Cc          : "wocmultimedia" info@wocmultimedia.com
Data      : Mon, 1 Oct 2012 06:36:05 -0700
Oggetto : Re: [maqetta] Mobile Theme disappears on reopening project (#3391)

**Comment by wocmultimedia** [comment]
I made this test:
Open M701 with latest project
I tried to reload the old project and it hangs
I deleted this project and project1 opened correctly
I went in file manager
chose one folder of the new projects and copied
1- file1.html then back to chrome/M701 and selected this project from the menu, it opened, no problem besides no pictures shown
2- add images folder from the old project to the new one and back to chrome/M701 and selected this project from the menu, it opened, no problem besides no pictures shown
I closed chrome/M701 and restart PC and server.bat and chrome/M701 anything open with no problems besides no pictures shown.
I had no special theme, it was set in origin how device-specific so no theme folder was present in the old project folder.
I suppose something could be wrong/corrupted in the previous folder structure or older folder was not compatible with new one. Anyhow, it seems that M701 solved the problem of missing pictures with new projects structure.
In case some anomalies happens, I will report it to you.
